src/test/ui/consts/const_fn_mut_refs.rs

```diff
@@ -0,0 +1,15 @@
+// run-pass
+
+#![feature(const_fn_mut_refs)]
+
+struct Foo {
+    x: i32
+}
+
+const fn bar(foo: &mut Foo) -> i32 {
+    foo.x + 1
+}
+
```

Comment:
Some tests with mutable borrows inside the function would also be good.

Reply:
Would also be good to have tests for other const contexts.

Reply:
How far can we go with this? AFAIK assignments are not allowed inside constants yet

Reply:
I believe they should be, especially if you have e.g. `let mut x = 0; x = 1;`

Reply:
https://github.com/rust-lang/rust/pull/66606/commits/0d41f749d17f3c2d62c33803712c01149bcd4d79 enabled this :P